api/provider/aws/test_aws/service_delete_test.go

```diff
@@ -0,0 +1,206 @@
+			Value:      "lb_id",
+		},
+	}
+
+	for _, tag := range tags {
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	taskARNs := []*string{
+		aws.String("arn:aws:ecs:region:012345678910:task/l0-test-svc_id1"),
+		aws.String("arn:aws:ecs:region:012345678910:task/l0-test-svc_id2"),
+	}
+
+	listTasksInput := &ecs.ListTasksInput{}
+	listTasksInput.SetCluster("l0-test-env_id")
+	listTasksInput.SetDesiredStatus(ecs.DesiredStatusRunning)
+	listTasksInput.SetStartedBy("l0-test-svc_id")
+
+	listTasksPagesFN := func(input *ecs.ListTasksInput, fn func(output *ecs.ListTasksOutput, lastPage bool) bool) error {
+		output := &ecs.ListTasksOutput{}
+		output.SetTaskArns(taskARNs)
+		fn(output, true)
+
+		return nil
+	}
+
+	mockAWS.ECS.EXPECT().
+		ListTasksPages(listTasksInput, gomock.Any()).
+		Do(listTasksPagesFN).
+		Return(nil)
+
+	describeServicesInput := &ecs.DescribeServicesInput{}
+	describeServicesInput.SetCluster("l0-test-env_id")
+	describeServicesInput.SetServices([]*string{
+		aws.String("l0-test-svc_id"),
+	})
+
+	deployment := &ecs.Deployment{}
+	deployment.SetId("l0-test-svc_id")
+	deployment.SetStatus(ecs.DesiredStatusRunning)
+
+	deployments := []*ecs.Deployment{
+		deployment,
+	}
+
+	service := &ecs.Service{}
+	service.SetDeployments(deployments)
+
+	services := []*ecs.Service{
+		service,
+	}
+
+	describeServicesOutput := &ecs.DescribeServicesOutput{}
+	describeServicesOutput.SetServices(services)
+
+	mockAWS.ECS.EXPECT().
+		DescribeServices(describeServicesInput).
+		Return(describeServicesOutput, nil)
+
+	for _, taskARN := range taskARNs {
+		stopTaskInput := &ecs.StopTaskInput{}
+		stopTaskInput.SetCluster("l0-test-env_id")
+		stopTaskInput.SetTask(*taskARN)
+
+		mockAWS.ECS.EXPECT().
+			StopTask(stopTaskInput).
+			Return(&ecs.StopTaskOutput{}, nil)
+	}
+
+	updateServiceInput := &ecs.UpdateServiceInput{}
+	updateServiceInput.SetDesiredCount(int64(0))
+	updateServiceInput.SetCluster("l0-test-env_id")
+	updateServiceInput.SetService("l0-test-svc_id")
+
+	updateServiceOutput := &ecs.UpdateServiceOutput{}
+
+	mockAWS.ECS.EXPECT().
+		UpdateService(updateServiceInput).
+		Return(updateServiceOutput, nil)
+
+	deleteServiceInput := &ecs.DeleteServiceInput{}
+	deleteServiceInput.SetCluster("l0-test-env_id")
+	deleteServiceInput.SetService("l0-test-svc_id")
+
+	deleteServiceOutput := &ecs.DeleteServiceOutput{}
+
+	mockAWS.ECS.EXPECT().
+		DeleteService(deleteServiceInput).
+		Return(deleteServiceOutput, nil)
+
+	target := provider.NewServiceProvider(mockAWS.Client(), tagStore, mockConfig)
+	if err := target.Delete("svc_id"); err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Len(t, tagStore.Tags(), 0)
+}
+
+func TestServiceDelete_idempotence(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
```

Comment:
A bug that @nehayward brought up - we should be also gracefully handling when the service doesn't exist period, e.g. it has no Layer0 tags.

Reply:
seems like a separate test case would be warranted for that? And why just service?

Reply:
Yeah should probably be a separate case. Never said just service?

Reply:
Adding a simple test case for when there aren't any tags. ✅